partner_tag_smart_assignation/models/res_partner_category.py

```diff
@@ -136,6 +141,17 @@ def get_partners_from_sql(self):
     def update_all_smart_tags(self):
         return self.search([("smart", "=", True)]).update_partner_tags()
 
+    @api.model
+    def _check_validity_dates(self):
+        """ Scheduled method to deactivate records past their validity date """
+        today = fields.Date.today()
+        records_to_deactivate = self.search([('valid_until', '<', today), ('active', '=', True), ('smart', '=', True)])
+        # Archive the tag and unlink the partner
+        records_to_deactivate.write({
+            'partner_ids': [(5, 0, 0)],
+            'active': False
+        })
```

Comment:
Could be called by the update_all_smart_tags cron method. What do you think?

Reply:
You're right